### PR TITLE
enh: rewrite user section, tweaks to other sections

### PR DESCRIPTION
**Pull request by drammock**
Here is a moderate update to the Users section, and minor tweaks to the other sections. Rendered version [here](https://github.com/drammock/expyfun/blob/gittin/git_flow.rst).  Some of these are bold edits so I'm fine if they don't fit your vision of what this document is meant to accomplish.  My goal was making the `Users` section as friendly as possible, and delaying some of the more technical details into the `Developers` section.

### CODE REVIEW

File: git_flow.rst [hunk 1]
```diff
@@ -11,71 +11,106 @@ code. Developers_ want to make some changes to the code. Maintainers_ are the
...
     $ cd expyfun
     $ python setup.py install
 
-Perhaps without realizing it, these users have set up a structure so that
-their local repository knows about a remote repository named "origin" that
-points to `<http://github.com/LABSN/expyfun.git>`_, and they have checked out
-the branch "master" from that repository. This can be seen by doing::
+This will create a folder ``expyfun`` in the current directory to store the
+source code, and install the necessary parts in the special places where python
+looks when you do :python:`import expyfun` and call ``expyfun`` functions in
```
Comment (Eric89GXL) [inline git_flow.rst]: if you want to do an inline `:python:` (which makes sense), you need to add a directive like the one I made for bash at the top of the file

**Comment by Eric89GXL** [comment]
My initial reaction was that it makes the document too long, but then I remembered that this stuff doesn't make any sense at first, so a more verbose/complete explanation is almost certainly better. I'll merge once you add the directive so that renders properly.

**Comment by drammock** [comment]
OK, directive added.  One question though: the Maintainers example talks about maintainers having push access to the upstream, but then the example code sets up the upstream with `git://` instead of `git@github`.  Was that an accident?  Should I fix that in this PR?

**Comment by Eric89GXL** [comment]
Yeah you're right that should be `git@github.com`

**Comment by drammock** [comment]
ok, fixed remote URLs in the example @Eric89GXL

**Comment by Eric89GXL** [comment]
Thanks for the fixes.

**Pull request merged**